src/main/java/com/microsoft/azure/util/AzureCredentialUtil.java

```diff
@@ -49,6 +49,16 @@ public static BaseCredentials getCredential(String credentialId) {
                 CredentialsMatchers.withId(credentialId));
         if (credential != null) {
             return credential;
+        }
+        credential = CredentialsMatchers.firstOrNull(
```

Comment:
what's the logic behind firstOrNull?

Reply:
It will use the matcher `CredentialsMatchers.withId(credentialId)` created to check the credentials of ` CredentialsProvider.lookupCredentials`, and it will return the first matched one. If there is no one matched, it will return null. `CredentialsMatchers.withId(credentialId)` matcher can match no more than one item.